### PR TITLE
Update COMPASS scripts to support python 3

### PR DESCRIPTION
**Pull request by xylar**
The changes mostly involve fixing `print` statements.

The list command has also been modified to produce a sorted list of test cases (as opposed to the random order that was previously produced).

### CODE REVIEW

**Comment by xylar** [comment]
@mark-petersen, could you make sure these changes work for you when using COMPASS as normal?

If so, we'll want to merge `develop` into `ocean/develop` once this is merged to `develop`.

**Comment by xylar** [comment]
@matthewhoffman, could you make sure land-ice test cases still work as expected with these updated versions of the scripts?

**Comment by xylar** [comment]
The intention is just to test that these don't break python 2.7 support for now.  Many other COMPASS scripts aren't ready for python 3 yet, so that testing will need to wait.

File: testing_and_setup/compass/manage_regression_suite.py [hunk 1]
```diff
@@ -123,13 +126,13 @@ def process_test_setup(test_tag, config_file, work_dir, model_runtime,
                 command)
 
             # Write test case run step
-            suite_script.write("print ' ** Running case {}'\n".format(
+            suite_script.write("print(' ** Running case {}'\n".format(
                 test_name))
             suite_script.write('try:\n')
             suite_script.write('    {}\n'.format(command))
-            suite_script.write("    print '      PASS'\n")
+            suite_script.write("    print('      PASS'\n")
```
Comment (mark-petersen) [inline testing_and_setup/compass/manage_regression_suite.py]: Missing right parenthesis. Same for all print statements in this file where only left parenthesis is added.

Comment (mark-petersen) [reply testing_and_setup/compass/manage_regression_suite.py]: fixed.

**Comment by mark-petersen** [comment]
Asked @matthewhoffman and he was satisfied with my ocean tests. Removing him as a reviewer.

**Comment by xylar** [comment]
@mark-petersen, sorry about that.  Thanks for the fixes.  Any reason I shouldn't go ahead and merge this to `develop`?

**Comment by mark-petersen** [comment]
@xylar, just coordinating timing. @mgduda is merging several small PRs right now, I'll do this one later today so we don't force him to redo his local merge.

**Comment by xylar** [comment]
Okay, glad I asked.  I'll leave it to you.

**Pull request merged**